projects/rhods/toolbox/rhods_update_datasciencecluster/tasks/main.yml

```diff
@@ -139,6 +139,17 @@
       loop_var: component
     retries: 60
     delay: 10
+  - name: Wait for the DSC object to be ready
+    shell:
+      # Looking at the status of individual components is not sufficient enough
+      # for determining the Ready status
+
+      oc get dsc/default-dsc -o json |
+      jq '.status.conditions[]|select(.type == "Ready" and .status == "True")'
```

Comment:
thanks @mcharanrm , the fix makes sense, just some nits:

* mind that the name of the DSC isn't always `default-dsc`
* the DSC status is easiest to catch at `.status.phase`
```
oc get dsc/{{ datasciencecluster_name }} -o jsonpath={.status.phase}
```
```
    until: dscluster_ready_status.stdout != "Ready"
```
* the task you wrote needed `set -o pipefail;`, otherwise, if the DSC doesn't exist, `set -o pipefail;` will fail but the error will be swallowed and Ansible will wait 10min for nothing

Reply:
Good catch! I copied it from a playbook I created locally to test the conditional statement and I forgot to change it   while updating the PR. 

Regarding set -o pipefail;, I was notified about it by ansible-lint and have already fixed it in the latest commits.

Reply:
Isn't it `dscluster_ready_status.stdout == 'Ready'` ?